### ISSUE TITLE
Add "Merge when all statuses are green" button

### ISSUE THREAD

**Issue by jrafanie** [arthurnn/meta#2]
It would be nice if you could automatically merge a pull request when all of the statues are green, similar to how gitlab does it:
https://about.gitlab.com/2015/12/22/gitlab-8-3-released/

**Comment by jrafanie** [comment]
If travis or any of the other "status api component" is having issues or just slow, it's really frustrating to review a pull request that's ready to be merged but not yet green.  I don't want to have to give a bot commit rights to my repos to do this for me when it's green.  It's a hack. Please fix this in github.

**Comment by jrafanie** [comment]
🙇 ❤️ Pretty please!

**Comment by arthurnn** [comment]
@sgrif asked me about status api the other day. I think he might want this feature too.

**Comment by sgrif** [comment]
Yeah, I was going to update the bot to do it but this would be a hugely useful feature in Github itself. Just think of the energy that could be saved by reducing the number of open browser tabs people will have!

**Comment by jrafanie** [comment]
Great points. My thing is I always re-review a PR to remember what it was before I merge, even if I already ack'd it. The time spent going through tabs and the 20 seconds of context switch requires me to think about it and takes me away from what I was doing when I remembered I wanted to merge some PRs waiting on Travis. 

I'm fortunate to also use gitlab for some repos and it's an amazing time saver to have this single feature.

**Comment by mullender** [comment]
We implemented this using a label and a webhook: add the label and the webhook merges when the build statuses are green. The webhook has to search for the correct PR though, because build statuses are tied to a Sha, not a pr
